app/org/apache/spark/deploy/history/SparkFSFetcher.scala

```diff
@@ -70,12 +69,7 @@ class SparkFSFetcher(fetcherConfData: FetcherConfigurationData) extends Elephant
    * properties in the configuration files. Triggering it too early will sometimes make the configuration object empty.
    */
   private lazy val _logDir: String = {
-    val conf = new Configuration()
```

Comment:
It's a great idea to remove the dependence on hadoop installation.
However, there remains a lot of work to do for this. For example org.apache.hadoop.conf.Configuration is used many times in the codes to detect hadoop version, find resource manager and job history server etc.
Actually, Dr. Elephant use filesystem from FileSystem.get(conf) to find the hdfs block size in HDFSContext.

WebHdfsFileSystem seems to be an old fashion class since I can't find it in hadoop2. If we want to get all the data just from web, the [HDFS Rest API](http://hadoop.apache.org/docs/stable/hadoop-project-dist/hadoop-hdfs/WebHDFS.html) may be a better choise. Then we can detect the active namenode just like the way we support RM HA.